tar/test/main.c

```diff
@@ -130,6 +130,13 @@ __FBSDID("$FreeBSD: src/usr.bin/tar/test/main.c,v 1.6 2008/11/05 06:40:53 kientz
 # include <crtdbg.h>
 #endif
 
+mode_t umasked(mode_t expected_mode)
+{
+	mode_t mode = umask(0);
+	umask(mode);
+	return expected_mode & ~mode;
+}
+
```

Comment:
Since the `mode` value here is constant for the entire test run, it would probably make more sense to compute it with the standard  `umask(umask_mode = umask(0))`  idiom in main.c (maybe just after the work directory is created around line 2920).  Then this function just becomes one line.